.github/scripts/create-file-index.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+import os
+import time
+import json
+import sys
+import gzip
+from datetime import datetime
+
+def list_files_with_metadata(directory, output_file):
+    # Record the start time
+    start_time = time.time()
+
+    # Get the current date and time for indexing
+    index_timestamp = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+
+    files_metadata = []
+
+    for root, dirs, files in os.walk(directory):
```

Comment:
FTR. walk seems to be already doing desired (right) thing and does not follow symlinked folders. We get

```
In [14]: list(os.walk('/tmp/1234'))
Out[14]: 
[('/tmp/1234', ['infinitum', 'subdir', 'linkgood'], []),
 ('/tmp/1234/subdir', ['subdir2'], ['file']),
 ('/tmp/1234/subdir/subdir2', [], ['file2'])]
```
for 
```
❯ tree /tmp/1234
/tmp/1234
├── infinitum -> /tmp/1234
├── linkgood -> subdir
└── subdir
    ├── file
    └── subdir2
        └── file2
```

note: we do not monitor empty folders below